src/browser/replay/replayMap.js

```diff
@@ -0,0 +1,191 @@
+import { spanExportQueue } from '../../tracing/exporter.js';
+
+/**
+ * ReplayMap - Manages the mapping between error occurrences and their associated
+ * session recordings. This class handles the coordination between when recordings
+ * are dumped and when they are eventually sent to the backend.
+ */
+export default class ReplayMap {
+  #map;
+  #recorder;
+  #exporter;
+  #api;
+  #tracing;
+
+  /**
+   * Creates a new ReplayMap instance
+   *
+   * @param {Object} props - Configuration props
+   * @param {Object} props.recorder - The recorder instance that will dump replay data
+   * @param {Object} props.exporter - The exporter that retrieves completed spans
+   * @param {Object} props.api - The API instance used to send spans
+   * @param {Object} props.tracing - The tracing instance to create contexts
+   */
+  constructor({ recorder, exporter, api, tracing }) {
+    if (!recorder) {
+      throw new TypeError("Expected 'recorder' to be provided");
+    }
+
+    if (!exporter) {
+      throw new TypeError("Expected 'exporter' to be provided");
+    }
+
+    if (!api) {
+      throw new TypeError("Expected 'api' to be provided");
+    }
+
+    if (!tracing) {
+      throw new TypeError("Expected 'tracing' to be provided");
+    }
+
+    this.#map = new Map();
+    this.#recorder = recorder;
+    this.#exporter = exporter;
+    this.#api = api;
+    this.#tracing = tracing;
+  }
+
+  /**
+   * Processes a replay by dumping it from the recorder, adding the replay ID attribute,
+   * and storing it in the map.
+   *
+   * @param {string} replayId - The ID to use for this replay
+   * @returns {Promise<boolean>} A promise that resolves to true if processing was successful
+   * @private
+   */
+  async _processReplay(replayId) {
+    try {
+      const context = this.#tracing.contextManager.active();
+      const recordingSpan = this.#recorder.dump(context, { clear: false });
+
+      if (!recordingSpan) {
+        console.warn('ReplayMap._processReplay: No recording span was created');
+        return false;
+      }
+
+      recordingSpan.setAttribute('rollbar.replay.id', replayId);
+      this.#exporter.export([recordingSpan.span]);
```

Comment:
When `recorder.dump()` calls `span.end()`, this is already done. https://github.com/rollbar/rollbar.js/blob/master/src/tracing/spanProcessor.js#L11-L14

Reply:
The doc refers to `exporter.export` here. That's my mistake.

Reply:
Right, because the act of ending the context makes the processor more export the span right? Oversight.

Reply:
> ending the span makes the span processor export the span right?

👍 Yes